keras_core/saving/legacy/saving_options.py

```diff
@@ -0,0 +1,24 @@
+import contextlib
+import threading
+
+
+class SaveOptionsContext(threading.local):
+    def __init__(self):
+        self.use_legacy_config = False
+
+
+_save_options_context = SaveOptionsContext()
+
+
+@contextlib.contextmanager
+def keras_option_scope(use_legacy_config=True):
+    use_legacy_config_prev_value = _save_options_context.use_legacy_config
```

Comment:
I think you can use `global_state.set_global_attribute()` for this instead of a `threading.local` subclass -- the benefit is that it centralizes global state management, making it easier to track

Reply:
Fixed! I've changed it to a simpler scope using global state attributes. Thanks!